File: aio-scheduler/common/response/rest_result.go
package response

import "aio-scheduler/common/constants"

type RestResult struct {
	Success bool        `json:"success"`
	Message string      `json:"message"`
	Data    interface{} `json:"data"`
}

func Success(data interface{}) *RestResult {
	return &RestResult{
		Success: true,
		Message: constants.Success,
		Data:    data,
	}
}

func Failure(message string) *RestResult {
	return &RestResult{
		Success: false,
		Message: message,
		Data:    nil,
	}
}

func FailureWithData(message string, date interface{}) *RestResult {
	return &RestResult{
		Success: false,
		Message: message,
		Data:    date,
	}
}

File: aio-backend/aio-service/src/main/java/com/stone0090/aio/service/model/web/protocal/RestResult.java
package com.stone0090.aio.service.model.web.protocal;

import lombok.Data;

import java.io.Serializable;

@Data
public class RestResult implements Serializable {

    private Boolean success;
    private Object data;
    private String message;
    //    private String errorCode;
    //    private Integer showType;
    //    private String traceId;
    //    private String host;

    public RestResult(Boolean success, Object data, String message) {
        this.success = success;
        this.data = data;
        this.message = message;
    }

    public static RestResult success() {
        return success(null);
    }

    public static RestResult success(Object data) {
        return new RestResult(true, data, null);
    }

    public static RestResult failure(String message) {
        return new RestResult(false, null, message);
    }

}
